src/interfaces/swig/Features.i

```diff
@@ -577,40 +458,7 @@ RANDOM_INTERFACE(RandomKitchenSinksDotFeatures)
 %include <shogun/features/MatrixFeatures.h>
 namespace shogun
 {
-#ifdef USE_BOOL
-    %template(BoolMatrixFeatures) MatrixFeatures<bool>;
-#endif
-#ifdef USE_CHAR
-    %template(CharMatrixFeatures) MatrixFeatures<char>;
-#endif
-#ifdef USE_UINT8
-    %template(ByteMatrixFeatures) MatrixFeatures<uint8_t>;
-#endif
-#ifdef USE_UINT16
-    %template(WordMatrixFeatures) MatrixFeatures<uint16_t>;
-#endif
-#ifdef USE_INT16
-    %template(ShortMatrixFeatures) MatrixFeatures<int16_t>;
-#endif
-#ifdef USE_INT32
-    %template(IntMatrixFeatures)  MatrixFeatures<int32_t>;
-#endif
-#ifdef USE_UINT32
-    %template(UIntMatrixFeatures)  MatrixFeatures<uint32_t>;
-#endif
-#ifdef USE_INT64
-    %template(LongIntMatrixFeatures)  MatrixFeatures<int64_t>;
-#endif
-#ifdef USE_UINT64
-    %template(ULongIntMatrixFeatures)  MatrixFeatures<uint64_t>;
-#endif
-#ifdef USE_FLOATMAX
-    %template(LongRealMatrixFeatures) MatrixFeatures<floatmax_t>;
-#endif
-#ifdef USE_FLOAT32
-    %template(ShortRealMatrixFeatures) MatrixFeatures<float32_t>;
-#endif
 #ifdef USE_FLOAT64
     %template(RealMatrixFeatures) MatrixFeatures<float64_t>;
```

Comment:
the only template that is actually used in the library

Reply:
yeah good. It is kind of strange that we instantiate all the templates even if not used ....
we should remove this one at some point as well, with a factory that has a similar signature as the ctor maybe?

Reply:
Yes, that should be doable, would have something like `create_features("MatrixFeatures", dims=(n_rows, n_cols), data=mat)`?